### ISSUE TITLE
TypeScript labeled tuples

### ISSUE THREAD

**Issue by esthedebeste** [DanielXMoore/Civet#534]
TypeScript supports inline labeled tuples, which do not change behavior, but do give developers a hint on the meaning of the values:
```ts
type Range = [start: number, end: number]
```
In Civet this same code compiles to `[{ start: number, end: number }]`, which has a different meaning.
[Playground link](https://civet.dev/playground?code=dHlwZSBSYW5nZSA9IFtzdGFydDogbnVtYmVyLCBlbmQ6IG51bWJlcl0%3D)

**Comment by GoToLoop** [comment]
Does this merge implementation takes into consideration optional named tuples?

type rgba = [r: number, g: number, b: number, a?: number]

**Comment by edemaine** [comment]
Ah, no; thank you. I'll fix that tomorrow.

**Comment by edemaine** [comment]
Should be all fixed in 0.6.18!

**Comment by esthedebeste** [comment]
TypeScript uses a slightly different spread syntax for labeled tuple types:
```ts
type Thingy = [first: number, ...rest: number[]] // ParseError
// equivalent to
type Thingy = [number, ...number[]] // no ParseError
```
